elephant/csd.py

```diff
@@ -1,3 +1,357 @@
+        F = simps(I, xlin)
+        return F
+
+    def integrate_3D(x, y, z, xlim, ylim, zlim, csd, xlin, ylin, zlin,
+                     X, Y, Z):
+        Nz = zlin.shape[0]
+        Ny = ylin.shape[0]
+        m = np.sqrt((x - X)**2 + (y - Y)**2 + (z - Z)**2)
+        m[m < 0.0000001] = 0.0000001
+        z = csd / m
+        Iy = np.zeros(Ny)
+        for j in range(Ny):
+            Iz = np.zeros(Nz)
+            for i in range(Nz):
+                Iz[i] = simps(z[:, j, i], zlin)
+            Iy[j] = simps(Iz, ylin)
+        F = simps(Iy, xlin)
+        return F
+    dim = 1
+    if ele_zz is not None:
+        dim = 3
+    elif ele_yy is not None:
+        dim = 2
+    x = np.linspace(xlims[0], xlims[1], res)
+    if dim >= 2:
+        y = np.linspace(ylims[0], ylims[1], res)
+    if dim == 3:
+        z = np.linspace(zlims[0], zlims[1], res)
+    sigma = 1.0
+    h = 50.
+    pots = np.zeros(len(ele_xx))
+    if dim == 1:
+        chrg_x = np.linspace(xlims[0], xlims[1], res)
+        csd = csd_profile(chrg_x)
+        for ii in range(len(ele_xx)):
+            pots[ii] = integrate_1D(ele_xx[ii], chrg_x, csd, h)
+        pots /= 2. * sigma  # eq.: 26 from Potworowski et al
+        ele_pos = ele_xx
+    elif dim == 2:
+        chrg_x, chrg_y = np.mgrid[xlims[0]:xlims[1]:np.complex(0, res),
+                                  ylims[0]:ylims[1]:np.complex(0, res)]
+        csd = csd_profile(chrg_x, chrg_y)
+        for ii in range(len(ele_xx)):
+            pots[ii] = integrate_2D(ele_xx[ii], ele_yy[ii],
+                                    x, y, csd, h, chrg_x, chrg_y)
+        pots /= 2 * np.pi * sigma
+        ele_pos = np.vstack((ele_xx, ele_yy)).T
+    elif dim == 3:
+        chrg_x, chrg_y, chrg_z = np.mgrid[xlims[0]:xlims[1]:np.complex(0, res),
+                                          ylims[0]:ylims[1]:np.complex(0, res),
+                                          zlims[0]:zlims[1]:np.complex(0, res)]
+        csd = csd_profile(chrg_x, chrg_y, chrg_z)
+        xlin = chrg_x[:, 0, 0]
+        ylin = chrg_y[0, :, 0]
+        zlin = chrg_z[0, 0, :]
+        for ii in range(len(ele_xx)):
+            pots[ii] = integrate_3D(ele_xx[ii], ele_yy[ii], ele_zz[ii],
+                                    xlims, ylims, zlims, csd,
+                                    xlin, ylin, zlin,
+                                    chrg_x, chrg_y, chrg_z)
+        pots /= 4 * np.pi * sigma
+        ele_pos = np.vstack((ele_xx, ele_yy, ele_zz)).T
+    pots = np.reshape(pots, (-1, 1)) * pq.mV
+    ele_pos = ele_pos * pq.mm
+    lfp = []
+    for ii in range(len(pots)):
+        rc = neo.RecordingChannel()
+        rc.coordinate = ele_pos[ii]
+        asig = neo.AnalogSignal(pots[ii], sampling_rate=pq.kHz)
+        rc.analogsignals = [asig]
+        rc.create_relationship()
+        lfp.append(asig)
+    # lfp = neo.AnalogSignalArray(lfp, sampling_rate=1000*pq.Hz, units='mV')
+    return lfp
+
+if __name__ == '__main__':
```

Comment:
I have commented it now.  I thought that this serves to illustrate how to access the functions, without needing another "demo" file.